passes/opt/opt_expr.cc

```diff
@@ -844,7 +858,7 @@ void replace_const_cells(RTLIL::Design *design, RTLIL::Module *module, bool cons
 			if (input.match("**")) ACTION_DO_Y(x);
 			if (input.match("1*")) ACTION_DO_Y(x);
 			if (input.match("*1")) ACTION_DO_Y(x);
-			if (consume_x) {
+			if (!keepdc) {
```

Comment:
@cliffordwolf I'm not sure what the significance of `consume_x` is?

Reply:
Apparently `consume_x` is part of a heuristic in `opt_expr`'s double-nested-loop that defers greedy x-consumption in case there's a better optimisation.

Reply:
Testcase that exhibits its need added in https://github.com/YosysHQ/yosys/pull/1994/commits/495acf981547850ac7ad5b765dd215b0c33d59ac -- `o == 1'b0` and `o == 1'bx` are both valid solutions, but the latter is preferable and `consume_x` enables this.